src/Features/CSharp/Portable/Snippets/CSharpConsoleSnippetProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Snippets
+{
+    [ExportSnippetProvider(nameof(ISnippetProvider), LanguageNames.CSharp), Shared]
+    internal class CSharpConsoleSnippetProvider : AbstractConsoleSnippetProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpConsoleSnippetProvider()
+        {
+        }
+
+        protected override bool ShouldDisplaySnippet(SyntaxContext context)
+        {
+            var csharpContext = (CSharpSyntaxContext)context;
+            var token = context.LeftToken;
+            var isInNamespace = token.GetAncestors<SyntaxNode>()
+                .Any(node => node.IsKind(SyntaxKind.NamespaceDeclaration) ||
+                             node.IsKind(SyntaxKind.FileScopedNamespaceDeclaration));
+
+            return csharpContext.IsStatementContext || (csharpContext.IsGlobalStatementContext && !isInNamespace);
```

Comment:
it seems odd that something could be in a GlobalStatementContext and also in a namespace.

Reply:
```
namespace Namespace;
$$
class Program
{
    public async Task MethodAsync()
    {
    }
}
";
```

I had this check specifically for this case.

Reply:
that's interesting.  i still view it as a bug that the context object says taht's a global-statement location.  that presumably actually causes issues for other features.  we should file an issue on this.

Reply:
did we fix this issue in another PR?  i seem to recall that happening?  can we undo this special casing now?

Reply:
Yes it's fixed. Removed this logic from the CSharpConsoleSnippetProvider entirely.